### ISSUE TITLE
Design.Portal UX discussion about the flow of activating account

### ISSUE THREAD

**Issue by pedrocasa** [Paratii-Video/paratii-portal#363]
**Issue to discuss decisions related to the activating account process.** 
-

User decision map  _Describe the process the user needs to do to get on his objective_ 

![37529070-ef52c71a-2914-11e8-9dac-4ca82f8bae51](https://user-images.githubusercontent.com/30123644/37543697-28fc0c2a-2941-11e8-8195-aa8bcf5dbca0.png)

**Situation** 
-
The decision map babove refers to a user that needs to get his vouchers activated and is using an anonymous account on his device. 

After the user clicks on the voucher input-form a modal would appear asking if he has already activated his account. After it appeared, he'll have three options that lead him to different flows.

**Flow 1**
-
 ( flow number one is the only one that can be secured in the middle of the other flows)
- He closes the modal 
- Go back to the last thing that was on the screen before the modal appears

**Flow 2**
- 
- The user has already secured his account
- He is capable to rewrite the 12 words
- He Rechoose a pin for the device ( we need to always explain that this PIN only works in this browser)
- He rewrites the PIN 
- He can do what he was supposed to before the modal appears (if he has enough PTI))

**Flow 3**
- 
- The user hasn't secured his account yet
- He learns about PTI account and what does it imply ( wallet-videos-steaking etc) **POSSIBLE TO JUMP THIS STEP**
- He learns how to secure his account on other computers and what is the security code ( 12w)
- The 12w appears to him
- He writes it somewhere 
- He rewrites it to confirm that he has wroten it correctly 
- He learns what is PIN
- He chooses a PIN
- He rewrites it
- He can do what he was supposed to before the modal appears (if he has enough PTI)

**Flow 4**
-

- The user account is already secure 
- He needs to upload a video
- "ENTER YOUR PIN" modal appears

**Questions**
-

- Should we use modals or pages in the flow number 2
- Should we use modals or pages in the flow number 3

**Comment by pedrocasa** [comment]
One possible solution for the flow 3 _( using modals on it)_
_It still needs to explain what is the PTI account and how it works somehow_


![paratii-login](https://user-images.githubusercontent.com/30123644/37494086-22d2e42e-2886-11e8-8bdf-ea7a1813eaf8.gif)

**Comment by felipegaucho** [comment]
Nice. The flow seems OK. 
What triggers the process - anything that requires an "activated" account? Or clicking on the top right icon?

#### On the flow / UX:
- I'm on the team for showing separate input boxes for each word of the seed, and a copy icon next to them all, that copies the whole thing to the clipboard.

- I think there should be a small space above the PIN keyboard, that displays, in "masked characters" (dots or Xs) how many of the 4 digits have been inputted.

#### Also, minor stuff (text, basically):
- Do you already have activated your account -> Do you already have an account?
Suggested subtitle: You have to do so in order to proceed with this action.
Options: I already have an account / I'm new here
- Security code (too similar to "security PIN") -> Your account seed
Suggested subtitle: This is the seed of your account. You can use it to access your funds and information from other devices.
_(as for the detailed text, we can review later)_
- Don't discard this 12 words -> Write down these 12 words in a safe place
- The icon is a bit meaningless to me, a paper being teared with a "12w"? Maybe a safe lock icon; or twelve little blocks listed in a paper piece with a lock on its corner; or a pencil, paper, and little blocks written in it...
- Enter your security PIN -> Create a security PIN.
Suggested subtitle: Use this pin to unlock your wallet in this device -> It will work like a password for your account, in this browser.
- Rewrite your security PIN -> Confirm security PIN.
Suggested subtitle: Rewrtite the 4 number security PIN to continue the process of unlocking the wallet (this is probably the most demotivating instruction an user could ever face 😆) -> Rewrite the PIN you just created. You'll have to remember it to move tokens around.

_(Why do each of these titles have a different colour? White and purple)_.

- On notification: Congratulations -> Welcome
Suggested subtitle: Your account have... -> Your account is activated. Check it on this icon up here to see details.

**Comment by bent0b0x** [comment]
Just a thought. Is it possible we could implement this entire flow without ever using the term wallet? If we just call it an account, the fact that there is a new authentication mechanism might not be as scary, it'll just be different. Advanced users might like seeing the term "wallet", but the vast amount of users will not understand it and the term "wallet" might make them think they are immediately giving away money or something.

**Comment by pedrocasa** [comment]
Based on Felipe's notes I've designed and re-designed the flows interfaces

**flow 1**
-
![01](https://user-images.githubusercontent.com/30123644/37835576-4251e368-2e8f-11e8-8050-d6805d09cb5d.gif)

**flow 2**
-
![02](https://user-images.githubusercontent.com/30123644/37835582-455a504a-2e8f-11e8-8bdd-3fed2b7004ef.gif)


**flow 3**
-
![03](https://user-images.githubusercontent.com/30123644/37835611-559f32ae-2e8f-11e8-9ad2-f700108dacd9.gif)

**Comment by jellegerbrandy** [comment]
Here are some more remarks:

* I am strongly against calling this "activating your account".  We can call it "securing your account" instead.
* Users know what "backup " means, and the 12 words are just that:  a backup of your account.  So let's say that explicitly.
*  We _know_ if the user has already secured his account or not - we do not need to ask.
* The question at the start of the flow should be more like "Before proceeding with the serious, we think you should secure this account. You have 3 options: (1) secure this account (2) you already _have_ created an account on another device, and want to use it here (3) no thanks, I'll secure it later.
* Option (1): we have the flow. Option (3) is easy. Option (2): @pedrocasa , do we have a screens for that?

**Comment by jellegerbrandy** [comment]
And just to be complete: we'd also need a single screen with the pinpad to unlock the wallet.

**Comment by pedrocasa** [comment]
The reason why  I tried to use activate instead of secure is that 

- We will need another way for the user to do it instead of only clicking on the "PTI needed trigger" because some users will already have the knowledge that some actions of our platform will need PTI to work and some of them will search for a place to create their account.  It's how most of the internet works if you need to pay for something or create your personal space the first thing that you'll do is passing through the bureaucratic step of telling your computer who you are. (this is not something that only regulars users do, we do all the time ... it's how the internet has educated us)

If we simply ignore that factor and show the user that he already has an account the first thing that he will think is probably "someone has used my computer and created an account on this strange website called Paratii" and we don't want to scare them with this, we have other processes that are quite complex and uncommon even for people that are crypto enthusiasts.

So in order to do it, we need a word that says to the user " this is the bureaucratic way of telling your computer who you are" because as I said before most of them will search for it before doing anything and maybe the action "secure your account" don't  represent it for the regular internet users. (Maybe the word activate may bother some of the most intelligent devs that have the knowledge that they have already an account but at least the regular internet users are not gonna be lost in our platform.)

**Comment by pedrocasa** [comment]
The reason why  I tried to use activate instead of secure is that 

- We will need another way for the user to do it instead of only clicking on the "PTI needed trigger". Some users will already have the knowledge that some actions of our platform will need PTI to work and some of them will search for a place to create their account.  It's how most of the internet works if you need to pay for something or create your personal space the first thing that you'll do is passing through the bureaucratic step of telling your computer who you are.  (this is not something that only regulars users do, we do all the time ... it's how the internet has educated us)


If we simply ignore that factor and show the user that he already has an account the first thing that he will think is probably "someone has used my computer and created an account on this strange website called Paratii" and we don't want to scare them with this, we have other processes that are quite complex and uncommon even for people that are crypto enthusiasts.

So in order to do it, we need a word that says to the user " this is the bureaucratic way of telling your computer who you are" because as I said before most of them will search for it before doing anything and maybe the action "secure your account" don't  represent it for the regular internet users. (Maybe the word activate may bother some of the most intelligent devs that have the knowledge that they have already an account but at least the regular internet users are not gonna be lost in our platform.)

**Comment by jellegerbrandy** [comment]
I'm all for simplifying, but I am against misrepresentation. 
We are not here asking the user to identify herself

What we are doing here is the first two steps of these 3 things:

- protect your account with a password (pin)
- backup your data (12 words)
- "tell your computer who you are" -> ask the user for email and name and picture

So it is not "tell your computer who you are ", it is also not "activate your account" (because it is working fine already). 

There is an ambiguity in the word "account" that may be confusing here. It often means "you and all yoru personal data", but in this case it really means something much more specific, "control over your ethereum address" or something. So perhaps we should look for a more specific word here?

**Comment by pedrocasa** [comment]
"I'm all for simplifying, but I am against misrepresentation.
We are not here asking the user to identify herself"

yes you are right. maybe I'm taking this too deep right now if no need. 
We should test before get to any conclusion ... after all, the only change would be in the verb

**Comment by bent0b0x** [comment]
I do think there is something to be said for @pedrocasa 's point about "activating your account". I cannot think of another example on the internet where you have an account created implicitly behind the scenes. There are some cases where you can use a product before signing up, but once you want to save anything you need to create an account, but it is clear that prior to that step you had no account.

In general I have another comment. As a user, I would likely be surprised that I could use the application _without_ creating an account. Personally, I would not trust that any of the videos I've uploaded will persist. 

I think keeping the verbiage as close to the status quo as normal is a good thing. Even if we misrepresent the fact that the user

**Comment by pedrocasa** [comment]
I've recreated the flow using the word "secure" instead of the word activate. The only thing I thought I would not need is to create a specific button for "no thanks, I'll secure it later."

here is the link https://xd.adobe.com/view/7ad2a905-db47-4e2a-a352-8ca75b7199e7/screen/0c15b956-c95e-447c-9c96-4c3ba3a6b616/flow-03-

But I still think we have to find a word that makes it easier for the user to get the impression that he does not have an active account the first time he accesses the portal. Maybe the word "secure" might make sense there but if the user is looking for this kind of process in a different way he will end up getting the impression that somehow he already has an active account, and he knew nothing about it and was not warned that it would happen.

**Comment by bent0b0x** [comment]
I do think there is something to be said for @pedrocasa 's point about "activating your account". I don't think people would really understand a service where an account is created implicitly behind the scenes. To me, keeping the verbiage as close to the status quo as normal is a good thing. Even if we may be slightly misrepresenting reality (i.e. an unencrypted wallet has already been created or the user already), as long as we don't compromise their security I'm not sure that it would be a bad thing if our ultimate goal is user adoption and not necessarily education. 

Open to another point of view though, these are just my initial thoughts.

**Comment by jellegerbrandy** [comment]
good points, I'm not sure if there is an utlimately right answer, but I do have some more remarks:

* Who's our target? This whole account stuff only comes into play mostly for users that are actually getting tokens. So these are not complete noobs. These are early adopters of a new platform that promises to use cryptomagic to give a better deal to creators. I think these people will be disappointed if we hide the tech too much (this is what @felipegaucho heard when he had a talk/walkthrough with the first testers). 
* Another target we are developing for are investors and supporters. 
* So our users are not your mom: they are video people that want to actively engage in a new crypto platform.
* The scope of this flow is this: the users need to choose a pin, and save the 12 words in a place where they can find them. If they do that, it is already enough. And we can force them to do both: the pin using the standard choose-and-confirm, the 12 words with the  trick of "did you write them down? Show it then by inserting them here".Point being: to get the noobs to do the right thing, we don't need to _explain_ anything. 
* We definitely do not want people who **do** understand to get suspicious or confused because we are oversimplifying
* This is a fact: the user has an ethereum address and wallet that is functional _before_ he does the security thing. (And we cannot lie about that)
* The  current "standard" of interacting with ethereum dapps and wallets is exemplfiied by MetaMask, We are already simplytinf/dumbing down a lot: let's not worry too much.

**Comment by felipegaucho** [comment]
Oh my, this seems on 🔥, prolix like hell, and we need to define answers for now.
And remember that, if it all goes right, this may become just a "login with uPort (or whatever) widget soon".

This is how MyEtherWallet **"properly"** warns its users (10 f* screens!):
<img width="866" alt="screen shot 2018-03-28 at 10 46 19 am" src="https://user-images.githubusercontent.com/26517120/38033704-fb9dd454-3276-11e8-9f72-9915cfa59187.png">

This is a thread that corroborates talks I've had w/ Jelle here in Brazil, on our security model:
<img width="647" alt="screen shot 2018-03-28 at 10 46 30 am" src="https://user-images.githubusercontent.com/26517120/38033732-106c7d7c-3277-11e8-9343-90a9f66571e8.png">

**Bottom line**: the model we're following is fine. Let's use "secure account" like Pedro's drawn on its new option (but please, check redundancy man, the subtitle is 100% redundant now - just say "It won't take more than a minute", or something like that).
I'd **love** if we called the seed phrase an "Account recovery key", instead. I feel this is the kind of experimentation that *must* be done for the ecosystem to find its own ways to become UX-friendlier.

**Comment by pedrocasa** [comment]
Links commented above and on https://github.com/Paratii-Video/paratii-portal/wiki/Design-Wiki have been updated according to the last comments of Felipe

**Comment by jellegerbrandy** [comment]
Guys, today I explained the whole thing to @LorenzoAlluminio (focusing more on the security aspect of what we are doing and less the UX).  

(With the current idea, we are sacrificing a bit of security by saving the mnemonic in the browser until the user "secures/actives" her account - typically, more safely, you would shwo the mnomomic phrase immediately on wallet generation and never save it)

So he suggested this: instead of "securing the existing account", we can also treat the "anonymous wallet" as a throwaway account, and create a new one instead.

This presents some technical complications, perhaps, but what about the UX? I seems it will help with the choice of terminology (we can say "create your account" again :-)). 

Plz discuss even more :-)

**Comment by bent0b0x** [comment]
To me, creating a new account when you need to start engaging in Txs will actually be more intuitive. I think if we can go this route on the technical side we should explore it.

**Comment by pedrocasa** [comment]
@jellegerbrandy 
"And just to be complete: we'd also need a single screen with the pinpad to unlock the wallet."
Done, updated on design wiki 

Flow 4
https://xd.adobe.com/view/7ad2a905-db47-4e2a-a352-8ca75b7199e7/screen/73c38251-0adc-495f-925b-4a57f604fc1e/flow-04

**Comment by felipegaucho** [comment]
Closing this for now, @jellegerbrandy feel free to ressurect it when you change your mind :)
